inprocess/src/main/java/io/grpc/inprocess/InProcessTransport.java

```diff
@@ -164,6 +167,14 @@ public InProcessTransport(
         Optional.<ServerListener>absent(), includeCauseWithStatus);
   }
 
+  public InProcessTransport(
```

Comment:
It is better practice to have the constructor with fewer arguments call the one with more arguments passing in the default for the extra arguments.  In this case, all of the existing logic moves to the one with assumedMessageSize and the one without assumedMessageSize would have no logic other than calling the one with it passing in a -1.

Reply:
I'm afraid we might need to change couple of things if we do this as there's one parameter which is used by InternalInProcess, "optionalServerListener" and is forwarded to the private constructor. If we do what has been suggested above then we specifically want users to provide "Optional.<ServerListener>absent()" when asking for assumedMessageSize in overloaded constructor. I tried couple of workarounds to just initialize this variable beforehand, but then we need to remove 'final' keyword from it as the warning says with assumption, it might have been initialized and the build fails.

Reply:
That was added in https://github.com/grpc/grpc-java/commit/fd5f4aac631b17ad8b8c3796a7b1502273b1424e . I think it is no longer being used that way. We can potentially delete some of that code, but we should double-check it isn't being used in this repo and not in google3.

Reply:
I think there is one mention of InternalInProcess in google3, [here](https://source.corp.google.com/piper///depot/google3/third_party/java_src/appengine_standard/jar_contents/runtime-impl.classes;l=8524;bpv=0;bpt=0) in runtime implementations. Not sure it is actually getting used.

Reply:
That isn't a usage. That is just a list of classes that are in the built output.

Reply:
What happened here? I don't see why we can't make the variable `final`.

There's only four pre-existing call sites for constructing an InProcessTransport, and three of them use the same constructor. Why can't we just modify the existing constructor and add `-1` to the two test lines that construct the transport?

Reply:
I was confused by this comment, but finally figured out the context.  The suggestion is to make `assumedMessageSize` final which means that it can't be initialized in the declaration, it must be initialized in all of the constructors.  So constructors can pass a -1 if they didn't have an assumedMessageSize argument.

Reply:
Marking this conversation as not resolved.

Reply:
#11605 this PR makes it possible to do the above mentioned thing. I was not sure about cleaning that code. Thanks.

Reply:
This was possible without that other PR. You just modify the existing constructor and change all callers.

Reply:
Okay so I lost the context why I kept it like that, I recall the problem is, there's a public constructor `InProcessTransport` which I was avoiding to touch as I thought we should not change the existing callers (what if clients are using that api directly, my concern was around this).

Reply:
New commit added with all the needful from this thread.

Reply:
While the constructor is public, the class is package-private, so the API is not actually exposed to arbitary callers.

Reply:
I see.